controller/space.go

```diff
@@ -66,18 +84,38 @@ func (c *SpaceController) Create(ctx *app.CreateSpaceContext) error {
 		*/
 
 		newArea := area.Area{
-			ID:      satoriuuid.NewV4(),
-			SpaceID: space.ID,
-			Name:    space.Name,
+			ID:      uuid.NewV4(),
+			SpaceID: rSpace.ID,
+			Name:    rSpace.Name,
 		}
 		err = appl.Areas().Create(ctx, &newArea)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create area: %s", space.Name))
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create area: %s", rSpace.Name))
 		}
 
 		res := &app.SpaceSingle{
-			Data: ConvertSpace(ctx.RequestData, space),
+			Data: ConvertSpace(ctx.RequestData, rSpace),
+		}
+
+		// Create keycloak resource for this space
+		resource, err := c.resourceManager.CreateResource(ctx, ctx.RequestData, rSpace.ID.String(), spaceResourceType, &rSpace.Name, &scopes, rSpace.OwnerId.String(), rSpace.Name+"-"+uuid.NewV4().String())
```

Comment:
It's a false premise that if the KC call fails it actually failed to create the thing in KC. :)

In general, a DB tx should be held as short as possible.

Is there any reason to not do the KC bit first? If that is all good, setup the stuff locally?

I'll ping you in the morning. There are about 100 'solutions' to these types of issues, but they all are in the category of; it depends..